sys/src/9/amd64/archamd64.c

```diff
@@ -67,16 +67,43 @@ cpuidinfo(uint32_t eax, uint32_t ecx, uint32_t info[4])
 	return 1;
 }
 
+static char *
+cpuidname(uint32_t *info0)
+{
+	char *vendorid;
+
+	if(memcmp(&info0[1], "GenuntelineI", 12) == 0) {
+		vendorid = "GenuineIntel";
+		return vendorid;
+	}
+	else if(memcmp(&info0[1], "AuthcAMDenti", 12) == 0) {
+		vendorid = "AuthenticAMD";
+		return vendorid;
+	}
+	else {
+		vendorid = nil;
```

Comment:
Why not just return strdup(&info[1]) here? Just because you do not know it does not mean it's invalid. It's unlikely but, well, you never know. Still, I like the approach.

I advocate the strdup instead of just returning &info[1] b/c we don't know what the caller might do with the passed-in data. It may not matter however.

Overall, for now, this looks fine.